generated/hdinsight/microsoft.hdinsight/2015-03-01-preview/types.md

```diff
@@ -24,6 +24,17 @@
 * **tags**: [ApplicationTags](#applicationtags): The tags for the application.
 * **type**: 'Microsoft.HDInsight/clusters/applications' (ReadOnly, DeployTimeConstant): The resource type
 
+## Resource Microsoft.HDInsight/clusters/extensions@2015-03-01-preview
+* **Valid Scope(s)**: ResourceGroup
+### Properties
+* **apiVersion**: '2015-03-01-preview' (ReadOnly, DeployTimeConstant): The resource api version
+* **clusterMonitoringEnabled**: bool (ReadOnly): The status of the monitor on the HDInsight cluster.
+* **id**: string (ReadOnly, DeployTimeConstant): The resource id
+* **name**: 'clustermonitoring' (Required, DeployTimeConstant): The resource name
+* **primaryKey**: string (WriteOnly): The cluster monitor workspace key.
+* **type**: 'Microsoft.HDInsight/clusters/extensions' (ReadOnly, DeployTimeConstant): The resource type
+* **workspaceId**: string: The cluster monitor workspace ID.
+
```

Comment:
This resource type is [almost a union discriminated by name, except one of the overlapping paths is parameterized](https://github.com/Azure/azure-rest-api-specs/blob/394395ed0bdc1c5c05cb37a031379ca03bfebbb5/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json#L37-L471). I don't think there's a good way to model this without some updates to the way discriminated unions are resolved in Bicep, so this resource should not be generated at present.